CHANGELOG.md

```diff
@@ -0,0 +1,163 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased] - 2025-01-21
+
+### Added
+- **New Prompt Extension Service**: Added comprehensive `PromptExtensionService` class with full API support
```

Comment:
I would say let's make the log way more concise and really the most important is to mention the `api` change for creating new extensions.